Source/WebCore/page/EventHandler.cpp

```diff
@@ -1082,14 +1083,18 @@ void EventHandler::updateSelectionForMouseDrag(const HitTestResult& hitTestResul
         // Reset base for user select all when base is inside user-select-all area and extent < base.
         if (rootUserSelectAllForMousePressNode && target->renderer()->positionForPoint(hitTestResult.localPoint(), HitTestSource::User, nullptr) < m_mousePressNode->renderer()->positionForPoint(m_dragStartPosition, HitTestSource::User, nullptr))
             newSelection.setBase(positionAfterNode(rootUserSelectAllForMousePressNode.get()).downstream(CanCrossEditingBoundary));
-        
+
+        Position newExtent;
         RefPtr rootUserSelectAllForTarget = Position::rootUserSelectAllForNode(target.get());
         if (rootUserSelectAllForTarget && m_mousePressNode->renderer() && target->renderer()->positionForPoint(hitTestResult.localPoint(), HitTestSource::User, nullptr) < m_mousePressNode->renderer()->positionForPoint(m_dragStartPosition, HitTestSource::User, nullptr))
-            newSelection.setExtent(positionBeforeNode(rootUserSelectAllForTarget.get()).upstream(CanCrossEditingBoundary));
+            newExtent = positionBeforeNode(rootUserSelectAllForTarget.get()).upstream(CanCrossEditingBoundary);
         else if (rootUserSelectAllForTarget && m_mousePressNode->renderer())
-            newSelection.setExtent(positionAfterNode(rootUserSelectAllForTarget.get()).downstream(CanCrossEditingBoundary));
+            newExtent = positionAfterNode(rootUserSelectAllForTarget.get()).downstream(CanCrossEditingBoundary);
         else
-            newSelection.setExtent(targetPosition);
+            newExtent = targetPosition.deepEquivalent();
+        if (newSelection.directionality() == Directionality::None && newExtent != newSelection.extent() && m_frame->selection().granularity() == TextGranularity::CharacterGranularity)
+            newSelection.setDirectionality(Directionality::None /*Weak*/);
```

Comment:
Doesn't this need to use `Directionality::Weak` in order to fix the issue? It seems the change is incorrectly left behind in the comment?

Reply:
Yes, that was an experiment because of some false positive failing tests. I need to change it back.